spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -895,6 +902,15 @@ else if (listener instanceof MessageListener) {
 				this.wantsFullRecords = false;
 				this.pollThreadStateProcessor = setUpPollProcessor(false);
 				this.observationEnabled = this.containerProperties.isObservationEnabled();
+
+				if (!AopUtils.isAopProxy(this.genericListener) &&
+					this.genericListener instanceof AbstractDelegatingMessageListenerAdapter<?>) {
```

Comment:
In the runtime, `KafkaBackOffAwareMessageListenerAdapter` is set for `this.genericListener` and `this.listener`. 
The `KafkaBackOffAwareMessageListenerAdapter` has reference of `RecordMessagingMessageListenerAdapter` as its `delegate`.

Please refer to a image below 🙇‍♂️ 
<img width="754" alt="image" src="https://github.com/user-attachments/assets/65613d8d-8118-4645-93a1-d67e4af850f2">

Reply:
Ok. What I mean that it might be better to have a generic solution. We with @sobychacko are working for async observation and gave sumilar condition almost at the same line of code. So, I’m anticipating to combine them in the end. Therefore this clarification is useful . Thanks

Reply:
So, I think your previous solution was correct.
We have to set callback only in case of a `KafkaBackoffAwareMessageListenerAdapter` which is indeed an indicator that we are in retry topic scenario.
This way we won't break all other use-cases where such an error handling would be unexpected.